python/lsst/daf/butler/registry/interfaces/_versioning.py

```diff
@@ -173,7 +177,7 @@ def tableSchemaRepr(table: sqlalchemy.schema.Table) -> str:
             tableSchemaRepr = [table.name]
             schemaReps = []
             for column in table.columns:
-                columnRep = f"COL,{column.name},{column.type}"
+                columnRep = f"COL,{column.name},{column.type.compile(dialect=dialect)}"
```

Comment:
I added this because (on a commit that's not on this branch) I switched to using a PostgreSQL-only `int8range` type for timespans, and the previous code here raised an exception because there was no stringification for that in the generic dialect that is used by default.  I think you're right that in other cases it makes no difference, except _maybe_ if a SQLAlchemy upgrade changed what dialect-specific type it mapped the a generic type to - but I imagine that would be considered a major breaking change in SQLAlchemy.